src/dachshund/scorer.rs

```diff
@@ -40,7 +40,7 @@ impl Scorer {
         }
     }
     // computes "cliqueness" score, i.e. the objective the search algorithm is maximizing.
-    pub fn score<TGraph: GraphBase>(&self, candidate: &Candidate<TGraph>) -> CLQResult<f32> {
+    pub fn score<TGraph: GraphBase>(&self, candidate: &mut Candidate<TGraph>) -> CLQResult<f32> {
```

Comment:
I'm not seeing anything here that should require the change to make candidate mutable, but we could leave this as a todo

Reply:
The checking of the local density threshold can update the candidate's guarantee.